main.go

```diff
@@ -18,18 +39,69 @@ func getPing(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func getBooks(w http.ResponseWriter, r *http.Request) {
+func Books(w http.ResponseWriter, r *http.Request) {
+	metodo := r.Method
+	fmt.Printf("metodo: %v\n", metodo)
+	switch {
+	case metodo == "GET":
+		getBooks(w, r)
+	case metodo == "POST":
+		postBooks(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
 
-	bookslist := []struct { //Struct created to handle data of the list of books
-		Name      string  `json:"name"`
-		Price     float32 `json:"price"`
-		Inventory int     `json:"inventory"`
-	}{
-		{"Book 1", 30.20, 2},
-		{"Book 2", 20.30, 1},
-		{"Book 3", 32.20, 5},
+func postBooks(w http.ResponseWriter, r *http.Request) {
+	//TO DO:
+	//Read the Json body
+	var newBook = Book{"sem nome", -1, -1, uuid.Nil}
```

Comment:
You don't need to manually initialize the values.
This is enough:
```suggestion
	var newBook Book

Reply:
Go makes sure that all the values are the zero value for their type on declaration.

Reply:
Yes, I know. But that is the way I've found of allowing me to check if the request message arrives complete.
It was important specially for the Inventory field, because (depending on the way that we will conceive the user workflow) a 'zero' value could not be a insertion fail, but a intended value. Could we keep this way or it's bad?

Reply:
We can express the absence of a field using a pointer. Maybe we should talk today.